src/services/products.js

```diff
@@ -0,0 +1,37 @@
+const { Product } = require('../models');
+
+class ProductsService {
+  async createProduct (req, res) {
+    const newProduct = await new Product(req.body).save();
+
+    return res.json(newProduct);
+  }
+
+  async updateProduct (req, res) {
+    const updateProduct = await Product.findByIdAndUpdate(
+      req.params.id,
+      req.body
+    ).exec();
+
+    return updateProduct ? res.json(updateProduct) : res.boom.notFound();
+  }
+
+  async getProducts (req, res) {
+    const getProducts = await Product.find(req.query)
```

Comment:
Don't pass entire original req.query object. You need to check what params are present... So we must be able find by next query params:
_id
name (regex)
category (_id)
size (_id)
status (_id)
minPrice, maxPrice (price range: min, max)